src/js/components/TaskView.js

```diff
@@ -60,6 +61,24 @@ class TaskView extends mixin(SaveStateMixin, StoreMixin) {
     super.componentWillMount();
   }
 
+  componentWillReceiveProps(nextProps) {
+    let {filterByStatus, searchString} = this.state;
+    let tasks = this.getFilteredTasks(filterByStatus, nextProps.tasks, searchString);
+    let prevTasks = this.getFilteredTasks(filterByStatus, this.props.tasks, searchString);
+
+    if (!deepEqual(tasks, prevTasks)) {
```

Comment:
I created DCOS-8883, DCOS-8884, and DCOS-8885 to address the underlaying issues in the TaskTable. This PR addresses the bug reported in DCOS-8414 and shouldn't be blocked by performance concerns as they are now to be tracked in those three tickets (: